gateway/src/main.rs

```diff
@@ -126,9 +127,17 @@ async fn start(db: SqlitePool, args: StartArgs) -> io::Result<()> {
 
     let api_handle = tokio::spawn(axum::Server::bind(&args.control).serve(api.into_make_service()));
 
-    let proxy = make_proxy(gateway, fqdn);
-
-    let proxy_handle = tokio::spawn(hyper::Server::bind(&args.user).serve(proxy));
+    let proxy = make_proxy(Arc::clone(&gateway), fqdn);
+    let (_resolver, tls_acceptor) = make_tls_acceptor();
+    let proxy_handle = if let UseTls::Disable = args.use_tls {
+        warn!("TLS is disabled in the proxy service. This is only acceptable in testing, and should *never* be used in deployments.");
+        axum_server::Server::bind(args.user).serve(proxy).boxed()
+    } else {
+        axum_server::Server::bind(args.user)
+            .acceptor(tls_acceptor)
+            .serve(proxy)
+            .boxed()
+    };
```

Comment:
Can we only initialise `tls_acceptor` in the `else` block since the `if` block does not use it? Ie moving line 131  to 135